gtest_parallel.py

```diff
@@ -358,6 +364,37 @@ def dump_to_file_and_close(self):
 
 # Record of test runtimes. Has built-in locking.
 class TestTimes(object):
+  class LockedFile(object):
+    def __init__(self, filename, mode):
+      self._filename = filename
+      self._mode = mode
+      self._fo = None
+
+    def __enter__(self):
```

Comment:
I was wrong above in "file locking will not work on Unix if TestTimes is used from multiple threads". It will not work only if threads share the same file descriptor. Since file descriptors are global within process then all threads will see the file lock for given file descriptor either as acquired or released. In our case TestTimes hides actual file descriptor and consequently it will work as expected.